operators/multiclusterobservability/controllers/multiclusterobservability/analytics/rightsizing_helper.go

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) Red Hat, Inc.
+// Copyright Contributors to the Open Cluster Management project
+// Licensed under the Apache License 2.0
+
+package analytics
+
+import (
+	"github.com/cloudflare/cfssl/log"
+	"gopkg.in/yaml.v2"
+)
+
+// FormatYAML converts a Go data structure to a YAML-formatted string
+func FormatYAML(data interface{}) string {
```

Comment:
maybe it can be `func FormatYAML[T RSPrometheusRuleConfig | clusterv1beta1.Placement](data T) string {`

Reply:
Updated the function parameters.